src/Compilers/Core/AnalyzerDriver/DeclarationInfo.cs

```diff
@@ -25,24 +23,24 @@ internal DeclarationInfo(SyntaxNode declaredNode, ImmutableArray<SyntaxNode> exe
             // Declared node is the identifier, which doesn't contain the initializer. Can we tweak the assert somehow to handle this case?
             // Debug.Assert(executableCodeBlocks.All(n => n.Ancestors().Contains(declaredNode)));
 
-            _declaredNode = declaredNode;
-            _executableCodeBlocks = executableCodeBlocks;
-            _declaredSymbol = declaredSymbol;
+            DeclaredNode = declaredNode;
+            ExecutableCodeBlocks = executableCodeBlocks;
+            DeclaredSymbol = declaredSymbol;
         }
 
         /// <summary>
         /// Topmost syntax node for this declaration.
         /// </summary>
-        public SyntaxNode DeclaredNode => _declaredNode;
+        public SyntaxNode DeclaredNode { get; }
 
         /// <summary>
         /// Syntax nodes for executable code blocks (method body, initializers, etc.) associated with this declaration.
         /// </summary>
-        public ImmutableArray<SyntaxNode> ExecutableCodeBlocks => _executableCodeBlocks;
+        public ImmutableArray<SyntaxNode> ExecutableCodeBlocks { get; }
 
         /// <summary>
         /// Symbol declared by this declaration.
         /// </summary>
-        public ISymbol DeclaredSymbol => _declaredSymbol;
+        public ISymbol? DeclaredSymbol { get; }
```

Comment:
Why is this annotated? The constructor has ` Debug.Assert(declaredNode != null);`

Reply:
This one is `declaredSymbol`, which can be null. `declaredNode` is the one that is guaranteed to be non-null.

Reply:
Whoops. Misread. Thanks

---
In reply to: [466720969](https://github.com/dotnet/roslyn/pull/46409#discussion_r466720969) [](ancestors = 466720969)